### ISSUE TITLE
Use 7zip format for storing artifacts

### ISSUE THREAD

**Issue by omus** [JuliaPackaging/BinaryBuilderBase.jl#110]
Based upon the [benchmarking](https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/106#issuecomment-769173686) done in #106 we should switch using the `7z` format with maximum compression and threading enabled. Unfortunately as [Pkg doesn't know about the `7z` format](https://github.com/JuliaLang/Pkg.jl/blob/76402b8974aba76f4412e59bee2e4bc6be1b7860/src/PlatformEngines.jl#L461-L465) support for that will need to be added first.

**Comment by omus** [comment]
Attempting to deal with this directly in Pkg: https://github.com/JuliaLang/Pkg.jl/pull/2375

**Comment by giordano** [comment]
Fixed by #113?

**Comment by omus** [comment]
Yes and no. We are now capable of using `7z`/`xz` as of #113 but we should make a plan for an opt-in or opt-out switch otherwise nothing will probably happen.
